### DISCUSSION TITLE
Нестабильное по времени открытие таблиц и разделов разница в 10 раз,долгая загрузка таблиц

### DISCUSSION THREAD

**Discussion by smplseven** [totumonline/totum-ru-issues-and-discussions#280] [Как это сделать?] [answered]
Система была развернута на одном хостинге, выявили эту проблему, кто инсталлировал систему сказал, что проблема не в коде системы. Перевезли систему на другой хостинг, сегодня начались такие же проблемы.

Сейчас:
Развернут сервер(SSD 180 GB NVMe , 4vCPUs,8GB RAM) 
Всего пользователей ~70,Онлайн в среднем 25 человек
4 основных раздела и 1 со статистикой 
В каждом  основном разделе по одной основной таблице и около 3-6 справочников 
В основных таблицах 2-3 тысяч строк на данный момент по 15-20 колонок  из них 5-10 колонок рассчитываются автоматически по справочнику
В одном из разделов у основной таблицы включено PROавтообновление

Проблема: 
Фризы периодические выходят при работе с системой, выглядят как на скринах одна и та же страница может загружаться 0,8 секунды или 8 секунды. Не зависит от географии пользователя или формата действий. Скрин нагрузки на железа тоже прилагаю, железо всё не загруженное совсем. 

![photo_2022-12-01_13-04-09 (2)](https://user-images.githubusercontent.com/113604604/205037874-a353bfc7-bcf2-429e-bb82-82f829cc3876.jpg)
![photo_2022-12-01_13-04-09](https://user-images.githubusercontent.com/113604604/205037880-84f81cde-840a-47e8-be64-aca874558613.jpg)
![photo_2022-12-01_13-04-10](https://user-images.githubusercontent.com/113604604/205037882-dca8e8da-926a-4e71-8175-c4bfd25323c8.jpg)
![photo_2022-12-01_13-58-49](https://user-images.githubusercontent.com/113604604/205037885-96de5026-d4ff-4b36-883c-e0b92b736576.jpg)
![photo_2022-12-01_13-58-59](https://user-images.githubusercontent.com/113604604/205037886-297c3a17-dfd2-4644-a2fe-967375643f5a.jpg)
![photo_2022-12-01_14-08-42](https://user-images.githubusercontent.com/113604604/205037888-063b077e-942d-4327-9adc-5e7d57c6f2ff.jpg)
![photo_2022-12-01_14-08-47](https://user-images.githubusercontent.com/113604604/205037890-2d1ec30c-5b04-4527-acf0-d4336c02c63a.jpg)
Из особенностей есть подключение напрямую к БД PowerBI, но система тормозит, даже когда запросов нет.

**Comment by k1erk** [comment, marked as answer]
А конфиг php-fpm покажите, а то пользователей у вас не мало. И посмотрите логи php8-fpm на наличие таких записей
WARNING: [pool www] server reached pm.max_children setting (5), consider raising it

**Reply by ttm-alexeypolunin** [reply]
Это первое, что надо сделать, так как у вас много оперативнки на сервере, а стандартный установщик настройку PM ставит на 2ГБ

https://ru.docs.totum.online/ubuntu#pm

Второе — это проверка одновременного доступа. Если у вас есть центральная таблица циклов и в циклах сидят 25 человек одновременно, то они будут втыкаться в актуальность таблицы циклов. Для таких случаев для таблицы Циклов в `Актуальность` ставится `Без отслеживания`.

Чего у нас не хватает, так это `Автообновления` подключенного к режиму `Без ослеживания`, сейчас `Автообновление` над `Нестрогим надстроено`. Запишемс.

**Reply by smplseven** [reply]
[totum]
pm = dynamic
pm.max_children = 100
pm.start_servers = 40
pm.min_spare_servers = 40
pm.max_spare_servers = 80
php_admin_value[display_errors] = stderr
php_admin_value[log_errors] = On
listen = /run/php/php8.0-fpm-totum.sock
listen.mode = 0660
listen.owner = totum
listen.group = www-data
user = totum
group = totum

php_admin_value[memory_limit] = 4096M
php_admin_value[upload_max_filesize] = 10M
php_admin_value[post_max_size] = 0
php_admin_value[opcache.enable] = 1
php_admin_value[opcache.enable_cli] = 1
php_admin_value[opcache.memory_consumption] = 512M
php_admin_value[jit_buffer_size] = 256M

php_admin_value[date.timezone] = Europe/Moscow

php_admin_value[session.gc_maxlifetime] = 3600
php_admin_value[session.save_path] = /var/lib/php/sessions_totum

Да, варнинги в логах  php8-fpm были ,настройки стояли везде минимальные ,сейчас скинул актуальные 
+отключили про.автобновление. Система стала работать гораздо лучше.

Основные таблица в разделах у нас простые ,только нагружены разными справочниками

**Reply by ttm-alexeypolunin** [reply]
Допрежимы в автообновлениях загрузим в течении пары дней.

**Reply by smplseven** [reply]
@k1erk скажите пожалуйста а по конфигам и настройке php-fpm всё нормально или нужно что-то изменить для быстродействия ?

**Reply by k1erk** [reply]
> @k1erk скажите пожалуйста а по конфигам и настройке php-fpm всё нормально или нужно что-то изменить для быстродействия ?

да вроде, все нормально. Но реальную картину вы увидите в работе.

**Reply by smplseven** [reply]
Спасибо!

**Reply by ttm-alexeypolunin** [reply]
> Допрежимы в автообновлениях загрузим в течении пары дней.

Добавили — там по названию понятно на каких режимах Актуальности они основаны.
